### PR TITLE
Added NodeStart event 

### PR DESCRIPTION
**Pull request by arendhil**
Added NodeStart event and implemented a working listener with debug on the Space example scene.

### CODE REVIEW

File: Unity/Assets/YarnSpinner/Scripts/DialogueRunner.cs [hunk 1]
```diff
@@ -127,8 +133,12 @@ public class StringUnityEvent : UnityEngine.Events.UnityEvent<string> { }
 
                     _dialogue.lineHandler = HandleLine;
                     _dialogue.commandHandler = HandleCommand;
-                    _dialogue.optionsHandler = HandleOptions;
-                    _dialogue.nodeCompleteHandler = (node) => { 
+                    _dialogue.optionsHandler = HandleOptions;
+					_dialogue.nodeStartHandler = (node) => {
+						onNodeStart?.Invoke(node, _dialogue.GetTagsForNode(node).ToArray());
```
Comment (desplesda) [inline Unity/Assets/YarnSpinner/Scripts/DialogueRunner.cs]: Can this be done without using LINQ?

Comment (arendhil) [reply Unity/Assets/YarnSpinner/Scripts/DialogueRunner.cs]: It'll not be as clean but it can be done. ^.^

Comment (arendhil) [reply Unity/Assets/YarnSpinner/Scripts/DialogueRunner.cs]: As a matter of fact, I think that this convertion from IEnumerable<string> to string[] is a waste, I'll just remove it and change the event associated to it.

**Comment by desplesda** [comment]
Thank you very much! I made some changes to make this line up a little better with existing API, and added documentation comments.

**Comment by arendhil** [comment]
Thanks, I'll take a look!

**Pull request merged**